mllib/src/main/scala/org/apache/spark/mllib/linalg/Matrices.scala

```diff
@@ -34,6 +34,9 @@ sealed trait Matrix extends Serializable {
   /** Number of columns. */
   def numCols: Int
 
+  /** Flag that keeps track whether the matrix is transposed or not. False by default. */
+  private[linalg] var isTransposed = false
```

Comment:
We should make this public because otherwise users won't be able to understand public members like `values`. This should be a `val` instead of `var` because we don't modify it in-place.